src/c/service.c

```diff
@@ -854,7 +857,33 @@ void devsdk_service_start (devsdk_service_t *svc, iot_data_t *driverdfls, devsdk
     iot_log_error (svc->logger, "Unable to load config file: %s", err->reason);
     return;
   }
+
   common_config_map = edgex_common_config_defaults (svc->name);
+
+  if (!svc->regURL)
+  {
+    common_config_file = edgex_device_loadConfig (svc->logger, svc->commonconffile, err);
+    if (err->code)
+    {
+      iot_log_error (svc->logger, "Unable to load common config file: %s", err->reason);
+      return;
+    }
+    iot_data_t *allservices_config_key = iot_data_alloc_string("all-services", IOT_DATA_REF);
+    iot_data_t *deviceservices_config_key = iot_data_alloc_string("device-services", IOT_DATA_REF);
+    iot_data_t *allservices_config = iot_data_map_get_map(common_config_file, allservices_config_key);
```

Comment:
You can just use `iot_data_string_map_get_map (common_config_file, "all-services");`
Also that string is #defined in consul.c, maybe move the definition to eg api.h?

Reply:
modified